route.go

```diff
@@ -89,34 +89,36 @@ func (r *Route) save() {
 	}
 }
 
-// Match check if the request match the route Pattern
-func (r *Route) Match(req *http.Request) bool {
+// Match check if the request matches the route Pattern and returns a map of the parsed variables
+// if it matches
+func (r *Route) Match(req *http.Request) (bool, map[string]string) {
```

Comment:
The match function, is not suppose to be called by a user, so even if you change the signature, nobody should be affected.